airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -354,3 +370,18 @@ def ti_put_rtif(
     _update_rtif(task_instance, put_rtif_payload, session)
 
     return {"message": "Rendered task instance fields successfully set"}
+
+
+def _is_eligible_to_retry(task_instance, task_retries: int):
+    """
+    Is task instance is eligible for retry.
+
+    :param task_instance: the task instance
+
+    :meta private:
+    """
+    if task_instance.state == State.RESTARTING:
+        # If a task is RESTARTING state it is always eligible for retry
+        return True
+
+    return task_retries and task_instance.try_number <= task_instance.max_tries
```

Comment:
Mainly a port over of https://github.com/apache/airflow/blob/088242a911ba52f16a8c97df0d6dcb7e47ca1b73/airflow/models/taskinstance.py#L1053-L1072 because we do not have "task_instance" table entries in SDK anymore.

Tried splitting it too because we do not have "task_instance.task" here